cdap-master/src/main/java/co/cask/cdap/data/tools/DatasetSpecificationUpgrader.java

```
@@ -0,0 +1,137 @@
+import org.apache.hadoop.hbase.HConstants;
+import org.apache.hadoop.hbase.client.HBaseAdmin;
+import org.apache.hadoop.hbase.client.HTable;
+import org.apache.hadoop.hbase.client.Put;
+import org.apache.hadoop.hbase.client.Result;
+import org.apache.hadoop.hbase.client.ResultScanner;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.NavigableMap;
+import java.util.SortedMap;
+import java.util.TreeMap;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Upgrading from CDAP version < 3.3 to CDAP version 3.3.
+ * <p>
+ * This requires updating the TTL property for DatasetSpecification in the DatasetInstanceMDS table.
+ */
+public class DatasetSpecificationUpgrader {
+  private static final Logger LOG = LoggerFactory.getLogger(DatasetSpecificationUpgrader.class);
+  private static final Gson GSON = new Gson();
+
+  private final HBaseTableUtil tableUtil;
+  private final Configuration conf;
+
+  @Inject
+  public DatasetSpecificationUpgrader(HBaseTableUtil tableUtil, Configuration conf) {
+    this.tableUtil = tableUtil;
+    this.conf = conf;
+  }
+
+  /**
+   * Updates the TTL in the {@link co.cask.cdap.data2.datafabric.dataset.service.mds.DatasetInstanceMDS}
+   * table for CDAP versions prior to 3.3.
+   * <p>
+   * The TTL for {@link DatasetSpecification} was stored in milliseconds.
+   * Since the spec (as of CDAP version 3.3) is in seconds, the instance MDS entries must be updated.
+   * This is to be called only if the current CDAP version is < 3.3.
+   * </p>
+   * @throws Exception
+   */
+  public void upgrade() throws Exception {
+    TableId datasetSpecId = TableId.from(Id.Namespace.SYSTEM.getId(), DatasetMetaTableUtil.INSTANCE_TABLE_NAME);
+    HBaseAdmin hBaseAdmin = new HBaseAdmin(conf);
+    if (!tableUtil.tableExists(hBaseAdmin, datasetSpecId)) {
+      LOG.info("Dataset instance table does not exist: {}. Should not happen", datasetSpecId);
```

Or it should throw an exception right?